### PR TITLE
feat(mojaloop/#2743): documentation standards + contributors guide on node upgrade recommendations

### PR DESCRIPTION
**Pull request by mdebarros**
feat(mojaloop/#2743): documentation standards + contributors guide on node upgrade recommendations - https://github.com/mojaloop/project/issues/2743
- Updates to `standards/creating-new-features`
   - Added `Working on your Feature` section to include steps (Dependency Upgrades, Vulnerability Checks, Node LTS upgrade) required for PRs to be accepted
   - Fixed linting issues
- Updates to `standards/guide`
   - Updated LTS references to `Active LTS`
   - Added `Runtime Environment` section including NodeJS, Docker, CI
   - Added `Naming Conventions` section
   - Enhanced `Directory Structure` section to include standard generic files and include nvmrc for stipulating NODE LTS version

docs: fixes to deployment guide
- Updated supported version for Mojaloop Helm releases v13 and v14
- Updated windows deployment guide to include an `RECOMMENDATIONS` for Aug 2022

chore: maintenance
- Added lint NPM scripts
- Added markdownlint.yaml configuration file
- Updated gitignore with standard ignores
- Updates for latest node LTS version
- Updated all dependencies
- Moved `Husky` git-hooks from package.json into Husky folder as per breaking changes introduced by upgraded dependency
- Updates to support ES-Module loader for latest `got` dependency lib
- Updated CI Config to use NVMRC file
- Replaced all occurrences of `{% uml src=... %}{% enduml %}` with a direct link to pre-generated SVG versions
- Regenerated all PUML -> SVG files due to the updated dependency libs

### CODE REVIEW

File: docs/community/standards/guide.md [hunk 1]
```diff
@@ -109,7 +241,7 @@ For other recommended bash conventions, refer to this blog post: [Best Practices
 
 ## Documentation
 
-- Documentation should be written in Mark Down format.  
+- Documentation should be written in Markdown format.
```
Comment (elnyry-sam-k) [inline docs/community/standards/guide.md:244]:
```suggestion
- Documentation should be written in Markdown format.  
- Hand drawn diagrams should use SVG format (example - architecture / component / block / state transition diagrams)
- Sequence diagrams should use PlantUML
```

File: docs/community/standards/guide.md [hunk 2]
```diff
@@ -109,7 +241,7 @@ For other recommended bash conventions, refer to this blog post: [Best Practices
 
 ## Documentation
 
-- Documentation should be written in Mark Down format.  
+- Documentation should be written in Markdown format.  
 - All discussion documents should be placed in /community/archive/discussion-docs.
 - The use of Google Docs and other tools should not be used.
```
Comment (elnyry-sam-k) [inline docs/community/standards/guide.md:246]:
```suggestion
- The use of Google Docs and other private tools is not advised for community wide collaboration
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -21,30 +21,44 @@ defaults_Dependencies: &defaults_Dependencies |
...
+    working_directory: &WORKING_DIR /home/circleci/project
     docker:
-      - image: node:12.16.1-alpine
+      - image: node:16.15.0-alpine # Ref: https://hub.docker.com/_/node?tab=tags&page=1&name=alpine
 
   legacy-docker:
-    working_directory: /home/circleci/project
+    working_directory: *WORKING_DIR
     docker:
       - image: node:10.15.2-alpine
```
Comment (elnyry-sam-k) [inline .circleci/config.yml:56]: should we leave this as it is?

**Review by elnyry-sam-k — DISMISSED** [review]

File: .circleci/config.yml (hunk 3 above)
Comment (mdebarros) [reply .circleci/config.yml:56]: It has to stay like that, unfortunately...It's one of the reasons we moved away from Gitbooks.

It was no longer supported or maintained. I.e. we can only build Gitbooks on Node v10 :(

File: docs/community/standards/creating-new-features.md [hunk 4]
```diff
@@ -67,13 +67,27 @@ It is recommended that `npm test` is executed after each of these steps to ensur
...
 
-2. REQUIRED - Audit Checks
+2. REQUIRED - Vulnerability Checks
 
    ```bash
    npm run audit:check
    ```
 
-   If there is no available working fix for the audit issue, you will need to do one of the following:
+   If required, one can execute [npm audit](https://docs.npmjs.com/cli/v8/commands/npm-audit) to apply any known available fixes:
```
Comment (elnyry-sam-k) [inline docs/community/standards/creating-new-features.md:76]:
```suggestion
[npm audit](https://docs.npmjs.com/cli/v8/commands/npm-audit) can be used to apply any known available fixes:
```

**Review by elnyry-sam-k — APPROVED** [review]

**Review by elnyry-sam-k — APPROVED** [review]

**Pull request merged**